recipe/build.sh

```diff
@@ -3,25 +3,14 @@
 
 if [[ $(uname) == 'Darwin' ]];
 then
-    export LIBRARY_SEARCH_VAR=DYLD_FALLBACK_LIBRARY_PATH
-
-    #############################################################
-    # Workaround for `libuuid` bug in 2.13.0.                   #
-    # Drop on subsequent `fontconfig` release.                  #
-    #                                                           #
-    # ref: https://bugs.freedesktop.org/show_bug.cgi?id=105366  #
-    #############################################################
+    # Use system UUID on macOS.
     export UUID_CFLAGS=" "
     export UUID_LIBS=" "
-    export PKGCONFIG_REQUIRES_PRIVATELY="${PKGCONFIG_REQUIRES_PRIVATELY} uuid"
-elif [[ $(uname) == 'Linux' ]];
-then
-    export LIBRARY_SEARCH_VAR=LD_LIBRARY_PATH
 fi
```

Comment:
My build hasn't finished yet, but I believe that if we patch `configure`, this entire `if` stanza should go away entirely. Otherwise, the `pkg-config` check for UUID succeeds because the environment variables take precedence over a missing `uuid.pc` file, and the way the new logic in `configure` works, that causes the `uuid` dependency in `Requires.private` to be inserted.

Reply:
Hmm...ok, I tried that initially, but the build failed. Will give it another try as we changed a few things since.

Reply:
Alright, worked that time. Maybe the build system wasn't being fully reconfigured before when I was having issues. Adding `autoreconf` likely fixed this. Pushed a fix.

Reply:
That makes sense: without `autoreconf`, we have to successfully run `configure` once in order to regenerate `configure`; without the environment variables, the first invocation fails and we never get to running the patched `configure` script. So `autoreconf` is in fact necessary here!

Reply:
PS. Fun times.